### PR TITLE
python3 compatibility for woth sorted-cmp as well as unicode

### PR DESCRIPTION
**Pull request by roaldatwork**
To fix the languages editor in admin application when running python 3

### CODE REVIEW

**Comment by roaldatwork** [comment]
I discovered that the translate functionality of the admin application doesn't work in python 3.
I found the gluon._compat.py file, and implemented the u() function according to [this topic](http://python3porting.com/noconv.html#more-bytes-strings-and-unicode)

**Comment by ilvalle** [comment]
isn't to_native enough? In addition, see comment on the code

File: gluon/_compat.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ def to_native(obj, charset='utf8', errors='strict'):
             return obj
         return obj.encode(charset, errors)
 
+    def u(x):
+        "safe alternative for the python 2 `unicode()` function, which doesn't exist in python 3"
+        return unicode('x', 'utf-8')
```
Comment (ilvalle) [inline gluon/_compat.py]: the `x` parameter is not used at all

Comment (roaldatwork) [reply gluon/_compat.py]: hmm fine by me as well. didn't know that one :)

Comment (roaldatwork) [reply gluon/_compat.py]: and whoops - i almost create a bug

Comment (roaldatwork) [reply gluon/_compat.py]: thats what we have peer reviews for - glad you catched it

**Comment by roaldatwork** [comment]
Sorry, I was to hasty to get the my fix in there. Fixed it now

File: gluon/_compat.py (hunk 1 above)
Comment (ilvalle) [reply gluon/_compat.py]: I suggest closing this PR and opening a new one that uses to_native without the `u()` method definition

Comment (roaldatwork) [reply gluon/_compat.py]: I followed up your advise. Thanks for the feedback

**Comment by roaldatwork** [comment]
I understand that - I'm too busy now, but will do that later today

**Pull request closed without merging**